manual/sphinx/user_docs/coordinates.rst

```diff
@@ -65,76 +68,108 @@ and the magnitudes of the reciprocal vectors are therefore
+   \nu\left(\psi, \theta\right) = \frac{{\boldsymbol{B}}\cdot\nabla\phi}{{\boldsymbol{B}}\cdot\nabla\theta} =
+   \frac{{B_{\text{tor}}}{h_\theta}}{{B_{\text{pol}}}R}
    \end{aligned}
+The sign of the poloidal field `{B_{\text{pol}}}` and toroidal field 
+`{B_{\text{tor}}}` can be either + or -.
 
-Where `\nu` is the local field-line pitch given by
+The field-aligned coordinates `\left(x,y,z\right)` are defined by:
 
 .. math::
+   :label: eq:coordtransform
 
    \begin{aligned}
-   \nu\left(\psi, \theta\right) = \frac{{\boldsymbol{B}}\cdot\nabla\zeta}{{\boldsymbol{B}}\cdot\nabla\theta} =
-   \frac{{B_{\text{tor}}}{h_\theta}}{{B_{\text{pol}}}R} = \frac{\left(F/R\right)h_\theta}{{B_{\text{pol}}}R} = FJ/R^2\end{aligned}
-
-where `F={B_{\text{tor}}}R` is a function only of `\psi`
-(sometimes called the poloidal current function).
+   x = {\sigma_{B\theta}}\left(\psi - \psi_0\right) \qquad y = \theta \qquad z = \sigma_{B\theta}
+   \left(\phi - \int_{\theta_0}^{\theta}\nu\left(\psi,\theta\right)d\theta\right)
+   \end{aligned}
 
 The coordinate system is chosen so that `x` increases radially
-outwards, from plasma to the wall. The sign of the toroidal field
-`{B_{\text{tor}}}` can then be either + or -.
+outwards, from plasma to the wall. The `y` coordinate increases in the
+same direction as `\theta` i.e. clockwise in the right-hand poloidal
+plane. The `z` coordinate increases in the same direction as `\zeta`
+i.e.  anti-clockwise looking from the top if `B_{pol}>0` and clockwise
+if `B_{pol} < 0`.
 
-The contravariant basis vectors are therefore
+This coordinate system is right-handed if `B_{pol}>0`, and left-handed if `B_{pol}<0`. 
+
+The reciprocal basis vectors are
 
 .. math::
 
    \begin{aligned}
-   \nabla x = {\sigma_{B\theta}}\nabla \psi \qquad \nabla y = \nabla \theta \qquad \nabla z =
-   {\sigma_{B\theta}}\left(\nabla\zeta - \left[\int_{\theta_0}^\theta{\frac{\partial \nu\left(\psi,
-   \theta\right)}{\partial \psi}} d\theta\right] \nabla\psi - \nu\left(\psi, \theta\right)\nabla\theta\right)\end{aligned}
-
+   \nabla x = {\sigma_{B\theta}}\nabla \psi \qquad
+   \nabla y = \nabla \theta \qquad
+   \nabla z = \nabla\zeta - \sigma_{B\theta}\left[\int_{\theta_0}^\theta{\frac{\partial \nu\left(\psi,\theta\right)}{\partial \psi}} d\theta\right] \nabla\psi
+   - \sigma_{B\theta}\nu\left(\psi, \theta\right)\nabla\theta
+   \end{aligned}
+  
 The term in square brackets is the integrated local shear:
 
 .. math::
 
    \begin{aligned}
    I = \int_{y_0}^y\frac{\partial\nu\left(x, y\right)}{\partial\psi}dy\end{aligned}
 
+  
+The basis vectors are:
+
+.. math::
+   
+   \begin{aligned}
+   \boldsymbol{e}_x =& J\left(\nabla y \times \nabla z\right) = {\sigma_{B\theta}} {\boldsymbol{e}}_\psi + I{\boldsymbol{e}}_\zeta \\
+   \boldsymbol{e}_y =& J\left(\nabla z \times \nabla x\right) = {\boldsymbol{e}}_\theta + \nu{\boldsymbol{e}}_\phi \\
+   \boldsymbol{e}_z =& J\left(\nabla x \times \nabla y\right) = {\boldsymbol{e}}_\zeta
```

Comment:
Following up to this point, I get a `\sigma_{B\theta}` everywhere in these basis vectors. Sorry! @bendudson did you maybe change a definition somewhere above?
```suggestion
    \boldsymbol{e}_x =& J\left(\nabla y \times \nabla z\right) = {\boldsymbol{e}}_\psi + {\sigma_{B\theta}} I{\boldsymbol{e}}_\zeta \\
   \boldsymbol{e}_y =& J\left(\nabla z \times \nabla x\right) = {\sigma_{B\theta}} \left({\boldsymbol{e}}_\theta + \nu{\boldsymbol{e}}_\phi\right) \\
   \boldsymbol{e}_z =& J\left(\nabla x \times \nabla y\right) = {\sigma_{B\theta}} {\boldsymbol{e}}_\zeta
```

Reply:
Thanks @johnomotani but I don't think that can be right: Where `I = 0`, surely `e_x = sigma_B e_psi` because e_x is outwards, while e_psi would be inwards if sigma_B < 0, and if the toroidal field is zero then `e_y = e_theta`

Reply:
For `e_x` I have:
`e_x = J (e^y x e^z) = h_theta / B_theta * ( e^theta x e^zeta - sigma_B * I * e^theta x e^psi)`

then `e^theta x e^zeta = |B_theta| / h_theta e_psi`  (note modulus, because (psi, theta, zeta) is right-handed) and `e^theta x e^psi = - |B_theta| / h_theta e_zeta`.

Reply:
There is definitely potential for confusion, because the `J` in this section is for the (x,y,z) coordinates, which is different (Bpol vs |Bpol|) from the J for the (psi, theta, zeta) system in the section above

Reply:
Ah, I think that would do it. So the `J` in the 'Field-aligned coordinates' section is for {x,y,z}, and is negative when the coords are left-handed, which they are when Bp<0. But the `J` in 'Orthogonal toroidal coordinates' is for {psi,theta,zeta} which are always right-handed so there J is +ve always. I think different symbols and a comment that their sign might be different would be helpful :smiley:

Reply:
Maybe `J_{\psi\theta\zeta}` and `J_{xyz}`?

Reply:
Ah yes that's a good idea. Thanks @johnomotani !

Reply:
Maybe I shouldn't be doing this so late, but is the `J` in the 'Magnetic field' section also the {x,y,z} one that's `J = \sigma_{B\theta} B_\mathrm{pol} / h_\theta`? Which would make `\boldsymbol{B} = \sigma_{B\theta} \frac{B_\mathrm{pol}}{h_\theta} \boldsymbol{e}_y, which was what we wanted to avoid? That seems to add up with `y=\theta` meaning y is always clockwise, so \nabla_y is in the opposite direction to Bp if Bp is negative (anticlockwise)... Maybe we should have `y = \sigma_{B\theta} \theta`? If we can/should do that it would have the advantage of x/y/z always being right-handed.

Reply:
Thanks John! Yes, in the next section I've written out the results for `y = \sigma_{B\theta} \theta`. Following the earlier confusion, I've re-labelled `y` to make `\eta = \sigma_{B\theta} \theta` for this right-handed coordinate system.